### PR TITLE
Devoxx day 1 part 1, keynote

### PR DESCRIPTION
**Pull request by JMesens**
We want to do an incremental blogpost with regular updates throughout the conference.
This is the first part with the keynote.

This is my first blogpost so I'm learning as I go, constructive feedback is highly appreciated.

### CODE REVIEW

File: _posts/2017-06-21-Devoxx-pl.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+---
+layout: post
+authors: [bart_blommaerts]
```
Comment (bart-blommaerts) [inline _posts/2017-06-21-Devoxx-pl.md]: That's not you :)
Look at data/authors.yml

File: _posts/2017-06-21-Devoxx-pl.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+title: "Devoxx Poland 2017"
+image: /img/2017-devoxx-pl/devoxx-poland.jpg
+tags: []
+category: Conference
+comments: true
+---
+# Devoxx Poland 2017
+(This post is still under progress and will be updated regularly)
+
```
Comment (bart-blommaerts) [inline _posts/2017-06-21-Devoxx-pl.md]: in progress

updated on a regular base?

File: _posts/2017-06-21-Devoxx-pl.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+## Discipline
+Automated verification is the software equivalent of exercising.
+
+It's 1820 all over again, patients died regularly within 3 weeks after operating.
+Doctors (Joseph Lister, Louis Pasteur) started cleaning his tools after surgery and noticed a positive trend in survival.
+
+>We're practicing a beautiful craft, let's go turn it into a wonderful profession. Focusing on quality and creative things.
+
+## Consumer Driven Contracts and Your Microservice Architecture
+By [Marcin Grzejszczak](https://twitter.com/@starbuxman) and [Josh Long](https://twitter.com/@starbuxman)
```
Comment (KevinVHoutte) [inline _posts/2017-06-21-Devoxx-pl.md]: Change Twitter handle

File: _posts/2017-06-21-Devoxx-pl.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+![Main room 1](/img/2017-devoxx-pl/room-1.jpg){: .image .fit }
+
+## Keynote: Speed without Discipline: a Recipe for Disaster 
+by [Venkat Subramaniam](https://twitter.com/@venkat_s)
+
+![Venkat Subramaniam - Speed without Discipline: a Recipe for Disaster](/img/2017-devoxx-pl/Venkat-Subramaniam-Speed-without-Discipline-a-Recipe-for-Disaster.jpg){: .image .fit }
+
+Paradigm shift right now
+- 90's -> object related programming (before procedures)
```
Comment (bart-blommaerts) [inline _posts/2017-06-21-Devoxx-pl.md]: In the nineties, everybody was doing imperative programming, using objects to implement functionality.

Maybe use full sentences?

File: _posts/2017-06-21-Devoxx-pl.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+- Latest -> imperative to declarative programming (functional programming, declerative thinking)
```
Comment (bart-blommaerts) [inline _posts/2017-06-21-Devoxx-pl.md]: declarative

File: _posts/2017-06-21-Devoxx-pl.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+## Declarative vs imperative
+
+* Manual driving -> imperative
+* Stick shift
+* Auto pilot in Tesla
+* Taxi with dedicated driver -> declarative 
+
+## JavaScript
+
+Black swan: ideal of concept you knew exists, but becomes unexpectedly prominent.
```
Comment (bart-blommaerts) [inline _posts/2017-06-21-Devoxx-pl.md]: of a concept that you knew, exists

File: _posts/2017-06-21-Devoxx-pl.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+Nobody expected JavaScript to become so wildly used 10 years ago.
+Venkat describes JavaScript as a slippery slope, never to use in the same sentence as logic.
+
+## Testing
+
+> Developing software without writing tests is described as JDD, Jesus Driven Development.
```
Comment (bart-blommaerts) [inline _posts/2017-06-21-Devoxx-pl.md]: "Pray that it works".

File: _posts/2017-06-21-Devoxx-pl.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+<blockquote class="twitter-tweet" data-lang="nl"><p lang="en" dir="ltr">Software development: a profession where people get paid to write poor quality code and get paid more later to cleanup the mess.</p>&mdash; Venkat Subramaniam (@venkat_s) <a href="https://twitter.com/venkat_s/status/648119106072387584">27 september 2015</a></blockquote>
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+> I like to automate my tests not because I have a lot of time, it is because I don't.
+
+We get confidence from automational testing to implement new features without even running the application, this is what we need to keep agile.
```
Comment (bart-blommaerts) [inline _posts/2017-06-21-Devoxx-pl.md]: automated testing

application. We need these to remain agile.

File: _posts/2017-06-21-Devoxx-pl.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+3. Right level to the right measure (test pyramid)
+
+![Testing ice-cream cone](/img/2017-devoxx-pl/testing-ice-cream-cone.png){: .image width="49%" }
+![Testing pyramic](/img/2017-devoxx-pl/testing-pyramid.png){: .image width="49%" }
+
+
+## Discipline
+Automated verification is the software equivalent of exercising.
+
+It's 1820 all over again, patients died regularly within 3 weeks after operating.
```
Comment (bart-blommaerts) [inline _posts/2017-06-21-Devoxx-pl.md]: where patients regularly died

File: _posts/2017-06-21-Devoxx-pl.md [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+Doctors (Joseph Lister, Louis Pasteur) started cleaning his tools after surgery and noticed a positive trend in survival.
```
Comment (bart-blommaerts) [inline _posts/2017-06-21-Devoxx-pl.md]: their tools

File: _posts/2017-06-21-Devoxx-pl.md (hunk 3 above)
Comment (JMesens) [inline _posts/2017-06-21-Devoxx-pl.md]: Juist, fixed. Thanks!

File: _posts/2017-06-21-Devoxx-pl.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+---
+# Devoxx Poland 2017
+_(This post is still in progress and will be updated on a regular base)_
+
+Krakow in the ICE Krakow Congress Centre.
+
+![ICE Krakow Congress Centre](/img/2017-devoxx-pl/venue-1.jpg){: .image height="420px" }
+![ICE Krakow Congress Centre foyer](/img/2017-devoxx-pl/venue-2.jpg){: .image height="420px" }
+
+We started off day 1 with the keynote in the absolutely phenomenal main room:
```
Comment (yannickdeturck) [inline _posts/2017-06-21-Devoxx-pl.md]: We started off day 1 with the keynote in the absolutely, phenomenal main room
(comma)

File: _posts/2017-06-21-Devoxx-pl.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+
+![Main room 1](/img/2017-devoxx-pl/room-1.jpg){: .image .fit }
+
+## Keynote: Speed without Discipline: a Recipe for Disaster ([Venkat Subramaniam](https://twitter.com/@venkat_s))
+
+Venkat kicked off the keynote, talking about a paradigm shift, that is happening right now in software development:
+In the nineties, everybody was doing **imperative programming**, using objects to implement functionality.
+Nowadays, this style of software development is shifting towards a more **declarative approach**.
+In imperative programming, developers focus on both **what** they want to do and **how** they want to do it. 
+In declarative programming, on the other hand, developers focus on **what** they want to do and use tools and libraries to facilitate their goal.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-Devoxx-pl.md]: In declarative programming on the other hand, developers ...
(delete the first comma)

File: _posts/2017-06-21-Devoxx-pl.md [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+let names = ["Dieter","Tom","Andreas","Ken","Yannick","Tim","Bart"];
+
+let count = 0;
+for(const name of names){
+  if(name.length === 4)
+    count++;
+}
+console.log(count)
+
+console.log(names.filter(name => name.lenght === 4).size)
```
Comment (yannickdeturck) [inline _posts/2017-06-21-Devoxx-pl.md]: name.lenght -> name.**length**

File: _posts/2017-06-21-Devoxx-pl.md [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+In the nineties, everybody was doing **imperative programming**, using objects to implement functionality.
+Nowadays, this style of software development is shifting towards a more **declarative approach**.
+In imperative programming, developers focus on both **what** they want to do and **how** they want to do it. 
+In declarative programming, on the other hand, developers focus on **what** they want to do and use tools and libraries to facilitate their goal.
+Venkat went on to state that programming in a functional style is declarative, but that not all declarative code is functional.
+
+	Functional style = declarative style + higher order functions
+
+{% highlight JavaScript %}
+let names = ["Dieter","Tom","Andreas","Ken","Yannick","Tim","Bart"];
```
Comment (yannickdeturck) [inline _posts/2017-06-21-Devoxx-pl.md]: Nitpicking, but I'd add a space after each comma.

File: _posts/2017-06-21-Devoxx-pl.md [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+console.log(count)
+
+console.log(names.filter(name => name.lenght === 4).size)
+{% endhighlight %}
+
+### Declarative vs Imperative
+
+Venkat told the audience that he doesn't like driving cars.
+He compared driving a stick shift to imperative programming.
+His goal is going from point A to point B and does not want to be involved in changing the gears _(Manipulating the DOM)_.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-Devoxx-pl.md]: ... and **he** does not want to be ...

File: _posts/2017-06-21-Devoxx-pl.md [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+> I automate my tests, not because I have a lot of time, but because I don't.
+
+After an introduction to declarative programming, Venkat switched to the topic of testing.
+To really be agile, we need to be confident that implementing new features won't cause failure.
+We can achieve this confidence by automating our tests and making sure they are repeatable.
+If we are really confident, we might even be able to ship software, without running the application.
+
+Writing software without writing tests is described as JDD: Jesus Driven Development. 
+Pray that it works.
+Obviously TDD (Test Driven Development) makes a lot more sense.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-Devoxx-pl.md]: Obviously, ...
(comma)

File: _posts/2017-06-21-Devoxx-pl.md [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+3. Right level of automation to the right measure (test pyramid)
+
+![Testing ice-cream cone](/img/2017-devoxx-pl/testing-ice-cream-cone.png){: .image width="49%" }
+![Testing pyramic](/img/2017-devoxx-pl/testing-pyramid.png){: .image width="49%" }
+
+
+### Discipline
+Automated verification is the software equivalent of exercising.
+
+It's 1820 all over again, patients died regularly within 3 weeks after operating.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-Devoxx-pl.md]: ... within **three** weeks ...

**Comment by Turbots** [comment]
Closing PR until the branch is intended to be published, otherwise it's confusing

**Pull request merged**